jvmhs/src/Jvmhs/Analysis/Reduce.hs

```diff
@@ -92,3 +93,100 @@ findUsedClasses  =
           (traverse.classNames)
           (traverse.classNames)
           nothing) cls
+
+
+chopVector :: Int -> V.Vector x -> V.Vector (V.Vector x)
+chopVector n vec
+  | divisible = chopVector' blockSize n vec V.empty
+  | otherwise = chopVector' blockSize (vectorLength `mod` n) vec V.empty
+  where blockSize = vectorLength `roundUpQuot` n
+        vectorLength = length vec
+        divisible =  length vec `mod` n == 0
+
+chopVector' ::
+     Int
+  -- ^ size of each Vec (delta)
+  -> Int
+  -- ^ num of vec remaining should be larger
+  -> V.Vector x
+  -> V.Vector (V.Vector x)
+  -> V.Vector (V.Vector x)
+
+chopVector' i n vec rslt
+  | V.null vec        = rslt
+  | otherwise         = let (block, rest) = V.splitAt newBlockSize vec
+                          in chopVector'
+                            newBlockSize newN rest (snoc rslt block)
+  where newN         = if n > -1 then n - 1 else n
+        newBlockSize = if n == 0 then i-1 else i
+
+roundUpQuot :: Int -> Int -> Int
+roundUpQuot i j = ceiling ((fromIntegral i / fromIntegral j)::Float)
+
+deltaI ::
+     V.Vector (V.Vector x)
+  -> Int
+  -> V.Vector x
+deltaI vecs i =
+  vecs V.! i
+
+deltaComplementI ::
```

Comment:
Maybe this operation is more efficient with sets, then the solution would just be: 

```haskell
[ world `S. difference` v | v <- vecs]
```

Reply:
I see your point! 
I think it is correct if we are only doing ddmin on class or (maybe? )on methods.
But if we need to do it on "code", we might need to remain the structure/order. Just like the html example on the paper, it keeps the relative position after each step.
Do we need to care about this situation?